MigrationAndImplementationGuide1-4.md

```diff
@@ -893,4 +893,166 @@ Now we can put these two sections back together to get the final message.
+                "object": "type2diabetes",
+                "predicate": "biolink:treats"
+            },
+        }
+    }
+    "results": [
+        {
+            "node_bindings": {
+                "query_node_0": [
+                    {
+                        "id": "type2diabetes"
+                        "query_id": "MONDO:0005015"
+                    }
+                ],
+                "query_node_1": [
+                    {
+                        "id": "metformin"
+                    }
+                ]
+            },
+            "edge_bindings": {
+                "query_edge_0": [
+                    {
+                        "id": "kg_edge_0"
+                    }
+                ]
+            }
+        }
+    ]
+}
+```
+
+## TRAPI 1.4 Example Message
+
+```
+"message": {
+    "query_graph": {
+        "nodes": {
+            "query_node_0": {
+                "ids": ["MONDO:0005148"]
+                "name": "Diabetes Mellitus"
+            },
+            "query_node_1": {
+                "categories": ["biolink:SmallMolecule"]
+            }
+        },
+        "edges": {
+            "query_edge_0": {
+                "predicates":["biolink:treats"],
+                "subject": "query_node_1",
+                "object": "query_node_0",
+                "knowledge_type": "lookup"
+            }
+        }
+    },
+    "knowledge_graph": {
+        "nodes": {
+            "diabetes": {node_info},
+            "type2diabetes": {node_info},
+            "metformin": {node_info}
+        },
+        "edges": {
+            "kg_edge_0": {
+                "subject": "metformin",
+                "object": "type2diabetes",
+                "predicate": "biolink:treats",
+            },
+            "kg_edge_1": {
+                "subject": "type2diabetes",
+                "object": "diabetes",
+                "predicate": "biolink:subclass_of",
+            },
+            "kg_edge_2": {
+                "subject": "metformin",
+                "object": "diabetes",
+                "predicate": "biolink:treats"
+                "attributes": [
+                    {
+                        "attribute_type_id": "infores:support_graphs",
+                        "values": [
+                            "a0"
+                        ]
+                    }
+                ]
+            },
+        }
+    },
+    "auxiliary_graphs": {
+        "subclass_explanation_1": {
+            "edges": [
+                "kg_edge_0",
+                "kg_edge_1"
+            ]
+        }
+    },
+    "results": [
+        {
+            "node_bindings": {
+                "query_node_0": [
+                    {
+                        "id": "diabetes"
+                    }
+                ],
+                "query_node_1": [
+                    {
+                        "id": "metformin"
+                    }
+                ]
+            },
+            "edge_bindings": {
```

Comment:
I don't think results have edge_bindings any more?

Reply:
Resolved

Reply:
I recall in the Architecture call 2 weeks ago, there was a question on whether KPs could use result.edge_bindings (particularly since they are often called to do one-hop edge-retrieval by ARAs). Was this question resolved? Does it depend on the subclassing discussion?